tools/ci/check_grep2.py

```diff
@@ -0,0 +1,337 @@
+            )
+
+
+def check_multiple_firelocks(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/machinery/door/firedoor[/\w]*?,\n[^)]*?/obj/machinery/door/firedoor[/\w]*?,\n[^)]*?/area/.+\)',
+            line,
+        ):
+            return (
+                "ERROR: Found multiple firelocks on the same tile, please remove them."
+            )
+
+
+def check_apc_pixel_shifts(filename, lines):
+    invalid_res = [
+        r"/obj/machinery/power/apc[/\w]*?[{]\n[^}]*?pixel_[xy] = -?[013-9]\d*?[^\d]*?\s*?[}],?\n",
+        r"/obj/machinery/power/apc[/\w]*?[{]\n[^}]*?pixel_[xy] = -?\d+?[0-46-9][^\d]*?\s*?[}],?\n",
+        r"/obj/machinery/power/apc[/\w]*?[{]\n[^}]*?pixel_[xy] = -?\d{3,1000}[^\d]*?\s*?[}],?\n",
+    ]
+    for line in lines:
+        for invalid_re in invalid_res:
+            if re.match(invalid_re, line):
+                return "ERROR: Found an APC with a manually set pixel_x or pixel_y that is not +-25. Use the directional variants when possible."
+
+
+def check_lattice_and_wall_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/lattice[/\w]*?,\n[^)]*?/turf/simulated/wall[/\w]*?,\n[^)]*?/area/.+?\)',
+            line,
+        ):
+            return "ERROR: Found a lattice stacked within a wall, please remove them."
+
+
+def check_window_and_wall_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/window[/\w]*?,\n[^)]*?/turf/simulated/wall[/\w]*?,\n[^)]*?/area/.+?\)',
+            line,
+        ):
+            return "ERROR: Found a window stacked within a wall, please remove it."
+
+
+def check_airlock_and_wall_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/machinery/door/airlock[/\w]*?,\n[^)]*?/turf/simulated/wall[/\w]*?,\n[^)]*?/area/.+?\)',
+            line,
+        ):
+            return "ERROR: Found an airlock stacked within a wall, please remove it."
+
+
+def check_grille_and_window_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/grille,\n[^)]*?/obj/structure/window/full[/\w]*?,\n[^)]*?/area/.+\)',
+            line,
+        ):
+            return "ERROR: Found grille above a fulltile window. Please replace it with the proper structure spawner."
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/grille,\n[^)]*?/obj/structure/window/full/plasmareinforced,\n[^)]*?/area/.+\)',
+            line,
+        ):
+            return "ERROR: Found grille above a fulltile plastitanium window. Please replace it with the proper structure spawner."
+
+
+def check_area_varedits(filename, lines):
+    for line in lines:
+        if re.match(r"^/area/.+[{]", line):
+            return "ERROR: Variable editted /area path use detected in a map, please replace with a proper area path."
```

Comment:
```suggestion
            return "ERROR: Variable edited /area path use detected in a map, please replace with a proper area path."
```

Reply:
Pulled the map checks but this will be in the following PR.